Rubrica/main.cpp

```diff
@@ -0,0 +1,232 @@
+#include <iostream>
+#include <vector>
+#include <string>
+
+using namespace std;
+
+struct Contact{
+    string nome;
+    string cognome;
+    string numero;
+};
+
+class IFilter {
+public:
+    virtual bool filter(const Contact &contact) = 0;
+};
+
+class NameStartsWith : public IFilter {
+    std::string m_name;
+public:
+    NameStartsWith(const std::string &name) : m_name(name) { };
+    bool filter(const Contact &contact) override {
+        if(m_name.at(0) == contact.nome.at(0)) return true;
```

Comment:
Così controlli solo la prima lettera del nome. Qualcosa di più preciso sarebbe controllare tutta la sottostringa `m_name`:
```suggestion
    return contact.nome.substr(0, m_name.size()) == m_name;
```
Per essere più precisi ancora dovremmo tenere di conto che `contact.name` può essere più corto di `m_name`.